analyzer/emerald/incoming.go

```diff
@@ -435,14 +435,12 @@ func emitRoundBatch(batch *storage.QueryBatch, qf *analyzer.QueryFactory, round
 	for key, change := range blockData.TokenBalanceChanges {
 		batch.Queue(qf.RuntimeTokenBalanceUpdateQuery(), key.TokenAddress, key.AccountAddress, change.String())
 	}
-	for tokenAddr, tokenData := range blockTokenData.TokenData {
-		batch.Queue(
-			qf.RuntimeTokenUpdateQuery(),
-			tokenAddr,
-			tokenData.Name,
-			tokenData.Symbol,
-			tokenData.Decimals,
-			tokenData.TotalSupply.String(),
-		)
+	for addr, possibleToken := range blockData.PossibleTokens {
+		if len(possibleToken.PossibleTypes) != 1 || possibleToken.PossibleTypes[0] != modules.EVMTokenTypeERC20 {
+			// We'll need to expand the discovery state to cover multiple
+			// possible token types, as we determine during scanning.
+			panic(fmt.Sprintf("discarding token %s type info %v", addr, possibleToken.PossibleTypes))
```

Comment:
We can probably simply dump the possible types into a SQL column? (Columns can be [array-typed](https://github.com/oasisprotocol/oasis-indexer/blob/77852199760eb6b7ae4a2ed7443950fbe9981868/storage/migrations/01_oasis_3_consensus.up.sql#L72-L72))

Regardless of whether you track this in the discover-state table or the tokens metadata table (I'm suggesting separating the two tables in another comment thread) or if you have a single table: At the end of the process, the tokens metadata table should contain the final determined type so we can present it to the frontend.

Reply:
trying without this info. may reconsider in future work on #225

Reply:
closing this out for now. added a cross reference from #225